modules/core/src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -331,73 +333,79 @@ private[kafka] object KafkaConsumer {
         F.guaranteeCase {
             requests.tryDequeue1
               .flatMap(_.map(F.pure).getOrElse(polls.dequeue1))
-              .flatMap(actor.handle(_) >> context.shift)
+              .flatMap(actor.handle(_))
               .foreverM[Unit]
           } {
-            case ExitCase.Error(e) => deferred.complete(Left(e))
-            case _                 => deferred.complete(Right(()))
+            case Outcome.Errored(e) => deferred.complete(Left(e)).void
+            case _                   => deferred.complete(Right(())).void
+            // TODO: cancelation?
           }
           .start
-          .map(fiber => Fiber[F, Unit](deferred.get.rethrow, fiber.cancel.start.void))
+          //.map(fiber => Fiber[F, Throwable, Unit](deferred.get.rethrow, fiber.cancel.start.void))
       }
     }(_.cancel)
 
   private[this] def startPollScheduler[F[_], K, V](
     polls: Queue[F, Request[F, K, V]],
     pollInterval: FiniteDuration
   )(
-    implicit F: Concurrent[F],
-    timer: Timer[F]
-  ): Resource[F, Fiber[F, Unit]] =
+    implicit F: Temporal[F]
+  ): Resource[F, Fiber[F, Throwable, Unit]] =
     Resource.make {
       Deferred[F, Either[Throwable, Unit]].flatMap { deferred =>
         F.guaranteeCase {
             polls
               .enqueue1(Request.poll)
-              .flatMap(_ => timer.sleep(pollInterval))
+              .flatMap(_ => F.sleep(pollInterval))
               .foreverM[Unit]
           } {
-            case ExitCase.Error(e) => deferred.complete(Left(e))
-            case _                 => deferred.complete(Right(()))
+            case Outcome.Errored(e) => deferred.complete(Left(e)).void
+            case _                 => deferred.complete(Right(())).void
           }
           .start
-          .map(fiber => Fiber[F, Unit](deferred.get.rethrow, fiber.cancel.start.void))
+         // .map(fiber => Fiber[F, Throwable, Unit](deferred.get.rethrow, fiber.cancel.start.void))
       }
     }(_.cancel)
 
   private[this] def createKafkaConsumer[F[_], K, V](
     requests: Queue[F, Request[F, K, V]],
     settings: ConsumerSettings[F, K, V],
-    actor: Fiber[F, Unit],
-    polls: Fiber[F, Unit],
+    actor: Fiber[F, Throwable, Unit],
+    polls: Fiber[F, Throwable, Unit],
     streamIdRef: Ref[F, Int],
     id: Int,
     withConsumer: WithConsumer[F]
-  )(implicit F: Concurrent[F]): KafkaConsumer[F, K, V] =
-    new KafkaConsumer[F, K, V] {
-      override val fiber: Fiber[F, Unit] = {
-        val actorFiber =
-          Fiber[F, Unit](F.guaranteeCase(actor.join) {
-            case ExitCase.Completed => polls.cancel
-            case _                  => F.unit
-          }, actor.cancel)
+  )(implicit F: Async[F]): F[KafkaConsumer[F, K, V]] = {
+     val actorFiber = F.uncancelable { _ =>
+          actor.join.flatMap { 
+            case Outcome.Succeeded(_) => polls.cancel
+            case _ => F.unit
+          }
+        }.onCancel(actor.cancel)
```

Comment:
when will this trigger, given that you've made it uncancelable? 
You've actually made it uncancelable in two different ways: not using `poll`, and `start`ing